src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -672,12 +673,63 @@ BinderCheckTypeResult ReportInvalidOperation()
 
         private static BinderCheckTypeResult CheckComparisonArgTypesCore(IErrorContainer errorContainer, TexlNode left, TexlNode right, DType typeLeft, DType typeRight)
         {
+            // Special case for number-backed BuiltIn Option Sets:
+            if (typeLeft.Kind == DKind.OptionSetValue || typeRight.Kind == DKind.OptionSetValue)
+            {                
+                var leftTypeDisambiguation = typeLeft.IsOptionSet && typeLeft.OptionSetInfo != null ? $"({typeLeft.OptionSetInfo.EntityName})" : string.Empty;
+
+                // Option set must be numeric to compare. 
+                if ((typeLeft.OptionSetInfo?.BackingKind ?? DKind.Number) != DKind.Number || (typeRight.OptionSetInfo?.BackingKind ?? DKind.Number) != DKind.Number)
+                {
+                    errorContainer.EnsureError(
+                        DocumentErrorSeverity.Severe,
+                        left.Parent,
+                        TexlStrings.ErrUnOrderedTypeForComparison_Type,
+                        typeLeft.GetKindString() + leftTypeDisambiguation);
```

Comment:
>leftTypeDisambiguation [](http://example.com/codeflow?start=51&length=22)

This error is also created if the error is on the typeRight. We probably want a different case here.